esphome/components/esp32_dac/output.py

```diff
@@ -2,14 +2,17 @@
 from esphome.components import output
 import esphome.config_validation as cv
 import esphome.codegen as cg
+from esphome.components.esp32 import get_esp32_variant
+from esphome.components.esp32.const import VARIANT_ESP32S2
 from esphome.const import CONF_ID, CONF_NUMBER, CONF_PIN
 
 DEPENDENCIES = ["esp32"]
 
 
 def valid_dac_pin(value):
+    valid_pins = (17, 18) if get_esp32_variant() == VARIANT_ESP32S2 else (25, 26)
     num = value[CONF_NUMBER]
```

Comment:
Something like this would be a little better and easier to understand
```suggestion
DAC_PINS = {
  VARIANT_ESP32: (25, 26),
  VARIANT_ESP32S2: (17, 18),
}

def valid_dac_pin(value):
    variant = get_esp32_variant()
    if (valid_pins := DAC_PINS.get(variant)) is None:
        raise cv.Invalid(f"DAC is not supported on {variant})
    num = value[CONF_NUMBER]
```

Reply:
Added, thanks